### DISCUSSION TITLE
Identifying dxf entities from exported svg elements

### DISCUSSION THREAD

**Discussion by LaDeKlint** [mozman/ezdxf#540] [Q&A] [answered]
I am doing off-line robot programming based on coordinates of dxf entities. I would  like to make an svg based web gui to make the user able to choose from different dxf entity coordinates to send to the robot.

1. Is there a way to identify the original dxf entity from an svg element in the resulting svg image when using the drawing/export add-on to export?

2. Or alternatively: is it possible to control the exact section (viewport) to export from the modelspace? This way I could transform dxf to svg coordinates and add interactive svg elements on top of the exported image.

A nod in the right direction or alternative solutions will be greatly apprieciated. 
Thank you.

**Comment by mbway** [comment, marked as answer]
1) It may be possible to pass metadata through matplotlib to the resulting svg by implementing a custom `matplotlib.backends.backend_svg.RendererSVG`. Using `matplotlib.artist.Artist.set_gid` you could implement a custom version of the `ezdxf.addons.drawing.matplotlib.MatplotlibBackend` which sets the id of each artist to some value which can be linked back to the dxf, then when your custom SVG renderer is writing out individual svg tags you may be able to obtain the id of the entity currently being drawn and be able to link that all the way back to the dxf entity id. But as you can see this won't be easy. An alternative would be to implement an `SVGBackend` to ezdxf directly. This is something which may be added to ezdxf in the future https://github.com/mozman/ezdxf/blob/master/TODO.md

btw I also checked whether it's possible to pass metadata via the Qt backend with [QSVGGenerator](https://doc.qt.io/qt-5/qsvggenerator.html), but I can't see any way you could get the metadata all the way from the ezdxf Qt backend (which draws to a `QGraphicsScene`) to the svg unlike with matplotlib where it *might* be possible. I suppose you could use `QSVGGenerator` together with a custom [QPaintEngine](https://doc.qt.io/qt-5/qpaintengine.html) to implement the `SVGBackend` but that might not give you much over just writing svg tags directly.

2) the export format being svg hopefully won't invalidate the solutions I have given in many previous questions:
- https://github.com/mozman/ezdxf/discussions/269
- https://github.com/mozman/ezdxf/discussions/219
- https://github.com/mozman/ezdxf/discussions/451
- https://github.com/mozman/ezdxf/discussions/357

@mozman do you think an FAQ or something in the documentation should be added to cover the 'how do i map between cad and the rendered image' because it seems like this question keeps being asked

**Reply by mozman** [reply]
A how-to section for the `drawing` add-on is on my todo list. But you can also do short posts in the "show and tell" section.

**Reply by LaDeKlint** [reply]
Thank you very much.
I'll give suggestion 2 a go.

**Reply by eevleevs** [reply]
Also interested, did you find a solution?
